core/src/main/java/com/orientechnologies/orient/core/record/impl/ODocument.java

```
@@ -584,6 +584,27 @@ public <RET> RET getPropertyWithoutValidation(String name) {
     return convertToGraphElement(value);
   }
 
+  @Override
+  public <RET> RET getPropertyOnLoadValue(String name) {
+    checkForLoading();
+
+    if (primaryRecord != null) {
+      return ((ODocument) primaryRecord).getPropertyOnLoadValue(name);
+    }
+    if (name == null) {
+      return null;
+    }
+
+    OVertexInternal.checkPropertyName(name);
+
+    var field = fields.get(name);
+    if (field != null){
+      return (RET) field.getOnLoadValue();
```

Please throw an exception in case the value returned is an ORidBag. To add a defense line to protect users from working with edges.